go/src/url_shortener/server.go

```diff
@@ -23,6 +24,40 @@ type URLShortener struct {
 	mux sync.Mutex
 }
 
+func (c *URLShortener) loadPersistenceFile(persistenceFile string) {
```

Comment:
In Go, it's best to always try to, when you can, pass arguments as general as possible.
Here `loadPersistenceFile` can only accepts data coming from the local file system. First that means that to test this function you're going to need to create a file on the filesystem, probably have this file committed in the repo, etc. 

For this use case it's best to declare the function as `loadPersistenceFile(r io.Reader)`. Why, this gives you the following advantages:
 - now you can load the data from anything that has a `Read` method, that means that `os.File` would work, but also a `bytes.Buffer`, a `net.Conn`, a `strings.Reader`, etc.
 - you're using streaming rather than having to load the whole file in RAM. What if the persistence file is hundreds of GB? streaming is very important in Go, not only because it's very elegant and makes code re-usable and composable. It's also very efficient. For instance, you could have the data stored on a gzip file, in this case you would create a gzip.Reader over an os.File, and pass that gzip.Reader to loadPersistenceFile which in turn will pass the reader to the json decoder. That means that you avoid unecessary copies and all the readers get composed together. `os.File -> gzip.Reader -> json.Decoder`. See https://godoc.org/encoding/json#ex-Decoder
 - testing would be easier, since you can directly create test json in the code, example: https://play.golang.org/p/25q2CMm0I_M